Production crash: NullPointerException in BalancePresenter.updateViewItems

Log

```
io.reactivex.exceptions.OnErrorNotImplementedException:
at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept (Functions.java:704)
at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept (Functions.java:701)
at io.reactivex.internal.observers.LambdaObserver.onError (LambdaObserver.java:77)
at io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onError (ObservableDoOnEach.java:117)
at io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext (ObservableDoOnEach.java:97)
at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal (ObservableObserveOn.java:201)
at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run (ObservableObserveOn.java:255)
at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run (HandlerScheduler.java:109)
at android.os.Handler.handleCallback (Handler.java:751)
at android.os.Handler.dispatchMessage (Handler.java:95)
at android.os.Looper.loop (Looper.java:154)
at android.app.ActivityThread.main (ActivityThread.java:6776)
at java.lang.reflect.Method.invoke (Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1496)
at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1386)
Caused by: java.lang.NullPointerException:
at io.horizontalsystems.bankwallet.modules.balance.BalancePresenter.updateViewItems (BalancePresenter.kt:147)
at io.horizontalsystems.bankwallet.modules.balance.BalancePresenter.access$updateViewItems (BalancePresenter.kt:16)
at io.horizontalsystems.bankwallet.modules.balance.BalancePresenter$viewDidLoad$1.accept (BalancePresenter.kt:39)
at io.horizontalsystems.bankwallet.modules.balance.BalancePresenter$viewDidLoad$1.accept (BalancePresenter.kt:16)
at io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext (ObservableDoOnEach.java:93)
```

--- a/app/src/main/java/io/horizontalsystems/bankwallet/modules/balance/BalanceModule.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/modules/balance/BalanceModule.kt
@@ -76,19 +76,23 @@ object BalanceModule {
         var items = listOf<BalanceItem>()
         var balanceSortType: BalanceSortType = BalanceSortType.Default
 
+        @Synchronized
         fun addUpdatedPosition(position: Int) {
             updatedPositions.add(position)
         }
 
+        @Synchronized
         fun clearUpdatedPositions() {
             updatedPositions.clear()
         }
 
         fun set(items: List<BalanceItem>) {
+            clearUpdatedPositions()
             originalItems = items
             sortBy(BalanceSortType.Default)
         }
 
+        @Synchronized
         fun getUpdatedPositions(): List<Int> = updatedPositions.distinct()
 
         fun getItem(position: Int): BalanceItem = items[position]

